Debug logger namespace standardization

scorbajio opened the pull request:
This change addresses the issues raised in #3689.

scorbajio commented:
The logger namespaces are now using lowercase tags.

trie:
```
  trie:find_path || Stack: ExtensionNode, BranchNode, BranchNode, BranchNode, LeafNode +0ms
  trie Setting root to 0x96ce81d93272f41719326f5cd4333c88f7ad403cc4404efc0fa62ffb10a88035 +2ms
  trie:put Key: 0x0c222c +0ms
  trie:put Value: 0x0c222c +0ms
  trie:find_path Target (6): [0,12,2,2,2,12] +0ms
  trie:find_path Walking trie from ROOT: 0x96ce81d93272f41719326f5cd4333c88f7ad403cc4404efc0fa62ffb10a88035 +0ms
  trie:lookup_node:by_hash 0x96ce81d93272f41719326f5cd4333c88f7ad403cc4404efc0fa62ffb10a88035 +0ms
  trie:lookup_node:by_hash ExtensionNode found in DB +0ms
  trie:find_path:extension_node Comparing node key to expected
  trie:find_path:extension_node || Node_Key: [0]
  trie:find_path:extension_node || Expected: [0]
  trie:find_path:extension_node || Matching: [true]
  trie:find_path:extension_node              +0ms
  trie:find_path:extension_node NextNode: 138,73,186,20,217,38,102,24,19,23,119,204,15,235,225,169,246,72,131,186,184,142,11,21,205,144,236,100,68,104,239,255 +0ms
  trie:lookup_node:by_hash 0x8a49ba14d9266618131777cc0febe1a9f64883bab88e0b15cd90ec644468efff +0ms
  trie:lookup_node:by_hash BranchNode found in DB +0ms
  trie:find_path:branch_node Looking for node on branch index: [12] +0ms
  trie:find_path:branch_node:12 NodeHash: 0xf2a8cf9296f999d637c17e03303af4aa4f0e5628d54d0e7ec3b6642015232f25 +0ms
  trie:lookup_node:by_hash 0xf2a8cf9296f999d637c17e03303af4aa4f0e5628d54d0e7ec3b6642015232f25 +0ms
  trie:lookup_node:by_hash BranchNode found in DB +0ms
  trie:find_path:branch_node Looking for node on branch index: [2] +0ms
  trie:find_path:branch_node:2 NodeHash: 0xcee1ccd3c512df214f62cf72006a64b394f6ffd73cfa55ef35e52711f6ef3b83 +0ms
  trie:lookup_node:by_hash 0xcee1ccd3c512df214f62cf72006a64b394f6ffd73cfa55ef35e52711f6ef3b83 +0ms
  trie:lookup_node:by_hash BranchNode found in DB +0ms
  trie:find_path:branch_node Looking for node on branch index: [2] +0ms
  trie:find_path:branch_node:2 Raw_Node: 32,44,99,114,101,97,116,101,32,116,104,101,32,108,97,115,116,32,98,114,97,110,99,104 +0ms
  trie:lookup_node:raw_node LeafNode +0ms
```

verkle:
```
  verkle:get Suffix: 1; Value: 0x320122e8584be00d000000000000000000000000000000000000000000000000 +0ms
  verkle:get Stem: 0x318dfa512b6f3237a2d4763cf49bf26de3b617fb0cabe38a97807a5549df4d; Suffix: 2 +0ms
  verkle:find_path Path (31): [0x318dfa512b6f3237a2d4763cf49bf26de3b617fb0cabe38a97807a5549df4d] +0ms
  verkle:find_path Starting with Root Node: [0x120744309fa260e19b58b2f5da88073fc62f883cd9cb9b0103d37d4002199802] +0ms
  verkle:find_path Partial Path 0x318d - found next node in path 0x41d4cf67efa1b8b71c7ae8345666433bf06ad7b9335642fc129122887a259e02. +0ms
  verkle:find_path Partial Path 0x318dfa51 - found next node in path 0x8225a4e1f63a9ed69ba4a486d5ce0d741cbb881054160d3b2f1f86941cab3406. +0ms
```

jochem-brouwer commented on packages/blockchain/src/blockchain.ts:
```diff
@@ -327,7 +327,10 @@ export class Blockchain implements BlockchainInterface {
-    const canonicalHeadHash = (await this.getCanonicalHeadHeader()).hash()
+    let canonicalHeadHash: Uint8Array | undefined
+    if (this.DEBUG) {
+      canonicalHeadHash = (await this.getCanonicalHeadHeader()).hash()
```
Ah nice find that this is only necessary when in debug mode :smile:

jochem-brouwer commented on packages/blockchain/src/blockchain.ts:
```diff
@@ -352,7 +355,7 @@ export class Blockchain implements BlockchainInterface {
       this.DEBUG &&
         this._debug(
-          `Canonical head set from ${bytesToHex(canonicalHeadHash)} to ${bytesToHex(hash!)} (number ${bigIntToHex(canonicalHead)})`,
+          `Canonical head set from ${canonicalHeadHash ? bytesToHex(canonicalHeadHash) : 'unknown'} to ${bytesToHex(hash!)} (number ${bigIntToHex(canonicalHead)})`,
```
Since `this.DEBUG` is true in this case, we know `canonicalHeadHash` is set, so the `? :` logic should be deleted here (if TypeScript complains, we know `canonicalHeadHash` is set so you can make TS know it is set by using `canonicalHeadHash!` (add exclamation mark))

scorbajio replied:
Fixed

jochem-brouwer commented on packages/verkle/README.md:
```diff
@@ -88,12 +88,13 @@ The `Verkle` class features optional debug logging. Individual debug selections
-| `verkle:*`          | verbose debug logging for all verkle methods     |
+| Logger              | Description                          |
+| ------------------- | ------------------------------------ |
+| `verkle:put`        | a verkle put operation has occurred  |
```
There should be an addition for `verkle:#` here.

scorbajio replied:
Fixed

jochem-brouwer commented on packages/client/README.md:
```diff
@@ -466,7 +466,7 @@ The following options are available:
 | Logger              | Description                                      |
 | ------------------- | ------------------------------------------------ |
-| `client:fetcher`            | This option enables logging for all fetchers      |
+| `client:fetcher:#`            | This option enables logging for all fetchers      |
```
I don't see any changes in the client code?

scorbajio replied:
I've updated the code.

jochem-brouwer reviewed (commented):
Some small nits

holgerd77 commented on packages/trie/README.md:
```diff
@@ -389,7 +401,7 @@ DEBUG=ethjs,trie:* npx vitest test/util/log.spec.ts
 Additional log selections can be added with a comma separated list (no spaces). Logs with extensions can be enabled with a colon `:`, and `*` can be used to include all extensions.
 
-`DEBUG=ethjs,tie:PUT,trie:FIND_PATH:* npx vitest test/proof.spec.ts`
+`DEBUG=ethjs,tie:put,trie:find_path:* npx vitest test/proof.spec.ts`
```
tie -> trie

scorbajio replied:
Fixed

jochem-brouwer commented on packages/blockchain/src/blockchain.ts:
```diff
@@ -352,7 +355,7 @@ export class Blockchain implements BlockchainInterface {
       this.DEBUG &&
         this._debug(
-          `Canonical head set from ${bytesToHex(canonicalHeadHash)} to ${bytesToHex(hash!)} (number ${bigIntToHex(canonicalHead)})`,
+          `Canonical head set from ${canonicalHeadHash} to ${bytesToHex(hash!)} (number ${bigIntToHex(canonicalHead)})`,
```
Should the `bytesToHex` be removed here?

scorbajio replied:
No, since it's going to be bytes. Fixed.

jochem-brouwer reviewed (commented):
One comment, LGTM

scorbajio commented:
I've updated the `trie` and `verkle` README's to be more accurate with namespace names, I've also added two missed debug symbols in `trie`: `from_proof` and `create_proof`. Lastly, I've went ahead and moved the branch index out of the namespace name and into the debug string, as discussed in #3689. An example:
```
  trie:#:lookup_node:by_hash 0x31e00526eadbd6cbfb83ca9000f2109f48d372a6a33dc4f9181ca4b43d146230 +0ms
  trie:#:lookup_node:by_hash BranchNode found in DB +0ms
  trie:#:find_path:branch_node Looking for node on branch index: [12] +0ms
  trie:#:find_path:branch_node Branch index: 12 - NodeHash: 0xd37ac31b9ede7380608d7fffe7a323fcafa87a264d06425ea8fbe7fa206ffdfb +0ms
```

acolytec3 reviewed (approved):
LGTM

acolytec3 reviewed (approved):
LGTM

jochem-brouwer commented:
Trie tests are failing?

scorbajio commented:
> Trie tests are failing?
The trie tests are failing on #3692 because printing node.value() for a trie node returns two possible types, a `Uint8Array` OR a `Uint8Array[]`, but only the former is visibly being returned since `this._value` which is typed as a `Uint8Array` is being returned. I saw that just printing the value of it to the debug string creates this array of numbers that isn't very accessible compared to printing hex, so I switched to using `bytesToHex` to print it, but when an array of u8s is the return value of `value()`, it type errors.

I can just go back to printing the plain value, but should we consider retyping `_value` and the return types of the trie node classes to make this stuff easier to see coming?

for reference see [node](https://github.com/ethereumjs/ethereumjs-monorepo/blob/15f8ff2bb119c89a1a5b835e48dbbd55e4705130/packages/trie/src/node/node.ts#L35)
